packages/server/src/distributed/redis/subservice.ts

```diff
@@ -84,8 +88,24 @@ export class RedisSubservice implements DistributedSubservice {
   }
 
   async destroy() {
-    await this.pub.quit()
-    await this.sub.quit()
+    const now = new Date()
+
+    for (const lock of Object.values(this.locks)) {
+      if (lock.expiry > now) {
+        try {
+          await this.release(lock)
+        } catch {
+          this.logger.error('Failed to release lock', _.omit(lock, 'lock'))
+        }
+      }
+    }
```

Comment:
Right now we hardcode every lock to have a max duration of 2 seconds. It would be possible in the future to have longer locks so this destroy() function that is executed when the server closes removes these locks. We aren't 100% precise with the timing by using new Date(), but it doesn't matter. The releasing of locks here is more for releasing locks that could still be alive for 30-40 secs.

Reply:
Why not destroy all locks no matter if they are expired or not, when we close the server? I think it would make more sense considering that you keep an in-memory reference of all the locks!

Reply:
We don't need to release expired locks. The locks contain the expiry on redis, so other nodes checking to get a lock will check the time if the already existing lock is expired.